sklearn/utils/estimator_checks.py

```diff
@@ -1475,22 +1475,25 @@ def check_estimators_overwrite_params(name, estimator_orig):
 
 
 @ignore_warnings(category=(DeprecationWarning, FutureWarning))
-def check_no_fit_attributes_set_in_init(name, Estimator):
-    """Check that Estimator.__init__ doesn't set trailing-_ attributes."""
-    # this check works on classes, not instances
+def check_no_attributes_set_in_init(name, Estimator):
```

Comment:
this should now take an estimator instance, not a class, since we've changed estimator checks altogether.